lib/spack/docs/packaging_guide.rst

```diff
@@ -2521,6 +2521,100 @@ would do with the package's own spec:
 
    mpicc = join_path(my_mpi.prefix.bin, 'mpicc')
 
+^^^^^^^^^^^^^^^^^^^^
+Spec build interface
+^^^^^^^^^^^^^^^^^^^^
+
+For more complex queries Spack supports an extended sub-scripting syntax:
+
+.. code-block:: python
+
+   mpi_cxx_and_fortran_query = spec['mpi:cxx,fortran']
+
+which allows for a list of comma separated values to be specified after
+the package name (a single colon separates the package name from the values).
+These values will be attached to the ``Spec`` object being created and can be
+used by the dependency to fine tune the computation of certain attributes:
+
+.. code-block:: python
+
+   # Get the list of MPI Fortran libraries
+   mpi_fortran_libs = spec['mpi:fortran'].libs
+
+   # Get the list of MPI C++ libraries
+   mpi_cxx_libs = spec['mpi:cxx'].libs
+
+The attributes that currently support this query mechanism are
+``libs`` and  ``cppflags``.
+
+``libs`` returns a ``LibraryList`` object. Its default
```

Comment:
[LibraryList](https://github.com/LLNL/spack/blob/develop/lib/spack/llnl/util/filesystem.py) is a wrapper around a list with a few convenience methods to manipulate libraries. It's been there since a while now. I think @adamjstewart is planning to add a `HeaderList` class (which I completely endorse).